core/domain/takeout_service_test.py

```diff
@@ -607,3 +650,36 @@ def test_export_data_nontrivial(self):
 
         exported_data = takeout_service.export_data_for_user(self.USER_ID_1)
         self.assertEqual(exported_data, expected_export)
+
+
+    def test_get_models_which_should_be_exported(self):
+        """Ensure that the set of models to export is the set of models with
+        export policy CONTAINS_USER_DATA, and that all other models have
+        export policy NOT_APPLICABLE.
+        """
+        all_models = []
+        model_names_list = [
+            model_name
+            for model_name in dir(models.NAMES)
+            if not model_name.startswith('__') and model_name != 'base_model'
+        ]
+        model_modules = models.Registry.import_models(model_names_list)
+        for model_module in model_modules:
+            for _, obj in inspect.getmembers(model_module):
+                if inspect.isclass(obj):
+                    all_models.append(obj)
```

Comment:
Hmmm. I don't think I quite understand. Isn't this a test of the function in takeout_service, so it should be in takeout_service_test?

Reply:
Hm, you shouldn't be duplicating code/functionality, though. I think it's fine to move it there, or (perhaps better) pull out the utility functions in storage_models_test and put them in a helper class in test_utils.py so that those utility functions can be referenced from both places.

@vojtechjelinek @varun-tandon what do you think?

Reply:
@seanlip I just tried swapping out this code with 
`all_models = models.Registry.get_all_storage_model_classes() `
and this seems to be working fine. I think I may have misunderstood why we need to move this function. Could you please indicate why this fix does not work?

Reply:
Oh, I didn't realize we had that function. That works too!

@vojtechjelinek -- is it worth having your code use that function?

Reply:
I created that function :smile: I use it in a few places.

Reply:
I think what should we do is to move the check that (almost) all models contain `get_export_policy` to `storage_models_test`, here you can have a method that will check that the `get_models_which_should_be_exported` contains all/some of the models that have `EXPORT_POLICY = CONTAINS_USER_DATA`

Reply:
Sorry, still a bit confused on this. Why do we want to move this test? What is wrong with keeping it here? And if we move it to storage_models_test, what would be left in this?

Reply:
Because `storage_models_test` contains tests to check if all the storage models classes contain the required methods like `get_deletion_policy` or `has_reference_to_user_id` it makes sense to move the test checking the presence of `get_export_policy`.

Reply:
Got it. Thanks